bin/image-create-openstack.sh

```diff
@@ -1,135 +1,40 @@
 #!/usr/bin/env bash
 
 # Checks and uploads specified Image to user's OpenStack account.
-# Downloads image first from an Amazon S3 storage account to host,
-# then uploads image to openstack teenancy.
-# Uses curl and python-glanceclient to do the job.
+#
+# 1. Downloads image first from an Amazon S3 storage (using curl and python-glance client)
+#
+# 2. Converts the image to the required format (if needed)
+#
+# 3. Uploads image to openstack tenancy. (using curl and python-glance client)
+#
+#
 #
 # Env vars
+#
 #   KN_IMAGE_NAME
+#   KN_DISC_FORMAT
 #   KN_IMAGE_BUCKET_URL
-
+#
 # Exit immediately if a command exits with a non-zero status
+
 set -e
 
+source ./bin/openstack/functions.sh
```

Comment:
Brilliant! We wanted to this a while ago, but we never had time. Maybe we could call the file `./bin/helpers/openstack.sh`? In this way we could have similar refactory for the other import scripts in the future (btw, if you feel like refactorying the other providers as well, it would be much appreciated).

Reply:
Hey, I've changed the filename as suggested.  I was thinking that it's a bit annoying having one provider use this pattern and the rest do not... but I don't have time to refactor the others for now anyways... thanks for the offer though ! :sweat_smile: